corgi/core/authentication.py

```diff
@@ -0,0 +1,76 @@
+from django.core.exceptions import ObjectDoesNotExist
+from mozilla_django_oidc.auth import OIDCAuthenticationBackend
+from rest_framework.permissions import BasePermission
+
+from corgi.core.models import RedHatUser
+
+
+class CorgiOIDCBackend(OIDCAuthenticationBackend):
+    """An extension of mozilla_django_oidc's authentication backend
+    which customizes user creation and authentication to support Red
+    Hat SSO additional claims."""
+
+    def verify_claims(self, claims):
+        """Require, at a minimum, that a user have a rhatUUID claim before even trying to
+        authenticate them."""
+        verified = super(CorgiOIDCBackend, self).verify_claims(claims)
+        return verified and "rhatUUID" in claims
+
+    def filter_user_by_claims(self, claims):
+        """The default behavior is to use e-mail, which may not be unique.
+        Instead, we use Red Hat UUID, which should be unique and persistent
+        between changes to other user claims."""
+        rhat_uuid = claims.get("rhatUUID")
+
+        if not rhat_uuid:
+            return self.UserModel.objects.none()
+
+        try:
+            rhat_user = RedHatUser.objects.get(rhat_uuid=rhat_uuid)
+            return [rhat_user.user]
+
+        except ObjectDoesNotExist:
```

Comment:
Should this be the more specific RedHatUser.DoesNotExist, or is there some other class (outside of the inheritance hierarchy) that can throw DoesNotExist errors we need to handle?
```suggestion
        except RedHatUser.DoesNotExist:
```